kratos/utilities/binbased_fast_point_locator.h

```diff
@@ -129,14 +130,13 @@ class BinBasedFastPointLocator
     {
         KRATOS_TRY
 
-        // Copy the elements to a new container, as the list will
-        //be shuffled duringthe construction of the tree
-        ContainerType& rElements = mrModelPart.ElementsArray();
-        IteratorType it_begin = rElements.begin();
-        IteratorType it_end = rElements.end();
-
-        typename BinsObjectDynamic<Configure>::Pointer paux = typename BinsObjectDynamic<Configure>::Pointer(new BinsObjectDynamic<Configure > (it_begin, it_end));
+        // Copy the entities to a new container, as the list will be shuffled duringthe construction of the tree
+        ContainerType entities_array;
+        entities_array= GetContainerType(mrModelPart, entities_array);
```

Comment:
wouldn't 
`ContainerType& entities_array = GetContainer(..)` suffice ?

Reply:
Nope, because that will complicate the template structure a lot, I need to pass by argument to avoid that